packages/grafana-ui/src/components/Dropdown/ButtonSelect.tsx

```diff
@@ -56,17 +57,20 @@ const ButtonSelectComponent = <T,>(props: Props<T>) => {
       </ToolbarButton>
       {isOpen && (
         <div className={styles.menuWrapper}>
-          <ClickOutsideWrapper onClick={onCloseMenu} parent={document}>
-            <Menu>
-              {options.map((item) => (
-                <MenuItem
-                  key={`${item.value}`}
-                  label={(item.label || item.value) as string}
-                  onClick={() => onChangeInternal(item)}
-                  active={item.value === value?.value}
-                />
-              ))}
-            </Menu>
+          <ClickOutsideWrapper onClick={onCloseMenu} parent={document} includeButtonPress={false}>
+            <FocusScope contain restoreFocus autoFocus>
+              <Menu onClose={onCloseMenu}>
+                {options.map((item) => (
+                  <MenuItem
```

Comment:
Hmm, for the RefreshPicker, I think `menuitemradio` is probably the more appropriate role, since an option is selected and persisted rather than just executing an action. I think that also means adding `aria-checked="true"` to the select option. Maybe we could add a `role` prop to `MenuItem`?

Reply:
Does changing the menuItem role to `menuitemradio` have any a11y impact?

Reply:
It depends on the screen reader software and settings used (for example I think this is the case for NVDA), but some will announce the role when focused on the menu item. "Menu item radio" lets users know that they are making a choice from a list of options that will be persisted in some way, which is the case here.

Reply:
Hmm..good point.
But isn't that similar to what I solved with the refresh Picker aria labels here? [run this branch to see how it works]

Reply:
Sure, it's similar, but I think in general it's better to go with established conventions as a user is more likely to expect and be familiar with those approaches. As well, if the choice is between an aria property with a clearer and more specific purpose vs. using, say, aria-label to fulfill a similar role, it's probably better to go with the first option, since it provides more information to screen readers.

Reply:
Good point.
I think in the case of refreshPicker, it makes sense to use both aria labels and aria role `menuitemradio`.

Reply:
would push some updates soon